src/hardware/pic.cpp

```diff
@@ -110,16 +118,17 @@ struct PIC_Controller {
 };
 
 static PIC_Controller pics[2];
-static PIC_Controller& master = pics[0];
-static PIC_Controller& slave  = pics[1];
+static PIC_Controller &primary_controller = pics[0];
+static PIC_Controller &secondary_controller = pics[1];
```

Comment:
Eh, these are a bit problematic… :(

master/slave terminology in here is totally stupid - primary/secondary is much better…

But these names were not freely chosen by author here - AFAIK it's reference to IBM's documentation :(

Suggestion: let's add documentation to the top of the file, based on this: https://en.wikipedia.org/wiki/Interrupt_request_(PC_architecture)#x86_IRQs - and in the end mention something like "Many historical documents use misleading "master/slave" terminology in reference to primary/secondary PIC".

This way code will be clear and people looking up IBM documentation won't need to rediscover that primary PIC was described as master PIC.

Reply:
Fixup inbound.